### PR TITLE
Add sphinx-based documentation

### PR DESCRIPTION
**Pull request by rw1nkler**
This PR replaces README-based documentation with a Sphinx-based version. The PR provides:
- CI tutorial tests (with tuttest)
- artifacts uploading in the CI
- sphinx docs directory
- RTD config

### CODE REVIEW

**Comment by rw1nkler** [comment]
This PR does not include the `simple_litex` demo. It will be included later.
(I have to rebase https://github.com/SymbiFlow/symbiflow-examples/pull/55 after merging this PR)

File: .github/scripts/build.sh
Comment (mithro) [inline .github/scripts/build.sh:8]: Why do we need this bit?

File: .github/scripts/build.sh
Comment (mithro) [inline .github/scripts/build.sh:21]: It seems like which examples are supported by which boards should be part of the documentation? Can we maybe use something like "selectors" with tuttest in some way?

File: .github/scripts/install-deps.sh
Comment (mithro) [inline .github/scripts/install-deps.sh:12]: This is needed for `tuttest` right? If so, can you be clear that is the case and this isn't used by the toolchain at all?

File: .github/scripts/install-deps.sh
Comment (mithro) [inline .github/scripts/install-deps.sh:32]: Why is `dataclasses` needed?

File: .github/scripts/install-deps.sh
Comment (mithro) [inline .github/scripts/install-deps.sh:1]: A lot of this feels like it should be in the documentation? IE Before getting started, make sure you have `git` and `wget` installed?

File: docs/building-examples.rst
Comment (mithro) [inline docs/building-examples.rst:16]: Could we maybe extend tuttest to understand `.. group-tab::` in some way?

File: docs/index.rst
Comment (mithro) [inline docs/index.rst:4]:
```suggestion
This guide explains how to get started with SymbiFlow and build example designs from the `SymbiFlow Examples <https://github.com/SymbiFlow/symbiflow-examples>`_ GitHub repository. It currently focuses on following FPGA families:
```

Fixed the URL and removed the `two` (that way when someone adds one more family, they don't have to change this).

**Review by mithro — COMMENTED** [review]
The documentation looks really good! Let's get this merged ASAP.

My only questions seem to be about the tuttest integration.

File: docs/building-examples.rst
Comment (rw1nkler) [reply docs/building-examples.rst:16]: Currently, we are supporting this directive. It is treated as a container for other directives. 
I believe that this is the correct approach, since `group-tab` can contain more directives. 

See: 
- https://docs.zephyrproject.org/latest/getting_started/index.html#install-dependencies
- https://github.com/zephyrproject-rtos/zephyr/blob/master/doc/getting_started/index.rst#install-dependencies

File: .github/scripts/install-deps.sh
Comment (rw1nkler) [reply .github/scripts/install-deps.sh:32]: I will fixed this with: https://github.com/antmicro/tuttest/pull/25

Comment (mithro) [reply .github/scripts/install-deps.sh:32]: Looks like that was merged!

**Comment by mithro** [comment]
Needs a rebase.

File: .github/scripts/build.sh
Comment (rw1nkler) [reply .github/scripts/build.sh]: We use it to "test the CI" locally, see: 
https://github.com/SymbiFlow/symbiflow-examples/pull/88/files#diff-7b3ed02bc73dc06b7db906cf97aa91dec2b2eb21f2d92bc5caa761df5bbc168fR45

File: .github/scripts/build.sh
Comment (rw1nkler) [reply .github/scripts/build.sh]: We added wildcards support to tuttest:
https://github.com/antmicro/tuttest/pull/26

File: .github/scripts/install-deps.sh
Comment (rw1nkler) [reply .github/scripts/install-deps.sh]: I extracted the tuttest from the docker script. So this part is no longer needed.

File: .github/scripts/install-deps.sh
Comment (rw1nkler) [reply .github/scripts/install-deps.sh]: This is available in the documentation: 
https://github.com/SymbiFlow/symbiflow-examples/pull/88/files#diff-f499103696248c617ff21a08e0a1a31cedab918185a5317a48c99c8997b22f54R14

File: .github/scripts/build-examples.sh
Comment (mithro) [inline .github/scripts/build-examples.sh:32]: Can you log a github issue to figure out a way to automatically determine these from the documentation? (Fixing not blocking merging.)

File: .github/scripts/build-examples.sh
Comment (mithro) [inline .github/scripts/build-examples.sh:1]: License header.

File: .github/scripts/common.sh
Comment (mithro) [inline .github/scripts/common.sh:1]: License header.

File: .github/scripts/install-toolchain.sh
Comment (mithro) [inline .github/scripts/install-toolchain.sh:1]: License header.

File: .github/workflows/sphinx-tuttest.yml
Comment (mithro) [inline .github/workflows/sphinx-tuttest.yml:10]: Can you log an issue to auto-generate this output? (Fixing not blocking merge)

File: .readthedocs.yml
Comment (mithro) [inline .readthedocs.yml:1]: License header.

File: README.rst
Comment (mithro) [inline README.rst:4]: Travis is no longer used right? This is all on GitHub actions now?

File: README.rst
Comment (mithro) [inline README.rst:22]: I would instead link to the rendered readthedocs output at `symbiflow-examples.rtfd.io`.

File: docs/Makefile
Comment (mithro) [inline docs/Makefile:1]: License header.

File: docs/conf.py
Comment (mithro) [inline docs/conf.py:1]: License header.

File: docs/conf.py
Comment (mithro) [inline docs/conf.py:65]: Pretty much all this should be moved into a separate `.py` file called from this file.

File: docs/conf.py
Comment (mithro) [inline docs/conf.py:83]: Would be good to have some short docstrings for all these functions.

File: docs/conf.py
Comment (mithro) [inline docs/conf.py:74]: Log an issue to have these auto-detected (Fixing not blocking merge).

File: .github/scripts/build-examples.sh
Comment (rw1nkler) [reply .github/scripts/build-examples.sh:32]: https://github.com/SymbiFlow/symbiflow-examples/pull/88#discussion_r541060922

File: .github/workflows/sphinx-tuttest.yml
Comment (rw1nkler) [reply .github/workflows/sphinx-tuttest.yml:10]: I opened the issue: https://github.com/SymbiFlow/symbiflow-examples/issues/94

File: docs/conf.py
Comment (rw1nkler) [reply docs/conf.py:64]: fixed

File: docs/conf.py
Comment (rw1nkler) [reply docs/conf.py]: added

File: docs/conf.py
Comment (rw1nkler) [reply docs/conf.py:20]: Added

File: docs/Makefile
Comment (rw1nkler) [reply docs/Makefile]: Added

File: .github/scripts/install-toolchain.sh
Comment (rw1nkler) [reply .github/scripts/install-toolchain.sh:1]: Added

File: .github/scripts/common.sh
Comment (rw1nkler) [reply .github/scripts/common.sh:1]: Added

File: .github/scripts/build-examples.sh
Comment (rw1nkler) [reply .github/scripts/build-examples.sh:1]: Added

File: docs/conf.py
Comment (rw1nkler) [reply docs/conf.py]: The issue is here: https://github.com/SymbiFlow/symbiflow-examples/issues/93

File: .readthedocs.yml
Comment (rw1nkler) [reply .readthedocs.yml:1]: fixed

File: README.rst
Comment (rw1nkler) [reply README.rst:4]: fixed

File: README.rst
Comment (rw1nkler) [reply README.rst:22]: fixed

**Comment by rw1nkler** [comment]
> Love the new images! Can you log an issue to get stabilized LED graphics? (Not blocking merging.)

Here is the issue: https://github.com/SymbiFlow/symbiflow-examples/issues/95

**Review by tmichalak — APPROVED** [review]
LGTM

**Pull request merged**
